kong/plugins/datadog/handler.lua

```diff
@@ -28,10 +28,11 @@ local get_consumer_id = {
 }
 
 
-local function compose_tags(service_name, status, consumer_id, tags, conf)
+local function compose_tags(service_name, status, consumer_id, tags, conf, route_name)
   local result = {
     (conf.service_name_tag or "name") .. ":" .. service_name,
-    (conf.status_tag or "status") .. ":" .. status
+    (conf.status_tag or "status") .. ":" .. status,
+    route_name and route_name ~= "" and ((conf.route_name_tag or "route") .. ":" .. route_name) or nil
```

Comment:
is it desidable that `route_name` is `nil` when not available or should it have some default? I'm thinking of what would happen if/when we add another value to this `result` array, then having a `nil` element in the middle of the array could affect iterations, although it seems we're passing this to generate a statsd_message, not sure how it would be handled there.

I'd like it better if we could keep this consistent with the other metrics and just do: `(conf.route_name_tag or "route") .. ":" .. route_name` here while passing some default value to this function to ensure `route_name` is not `nil`, wdyt?

Reply:
Not sure what we can set for default, it might be confusing.

Maybe the route ID instead?

Reply:
@samugi Okay the logic is now:

- route must actually be present - nil check
- if route.name is there: gsub the dots away for _
- if route.name is not there, use route.id
- otherwise route is nil for some reason, fall back to empty string

That should do it?